wit/deps/rdbms/postgres.wit

```diff
@@ -0,0 +1,296 @@
+  }
+
+  variant int8bound {
+    included(s64),
+    excluded(s64),
+    unbounded
+  }
+
+  variant numbound {
+    included(string),
+    excluded(string),
+    unbounded
+  }
+
+  variant tsbound {
+    included(timestamp),
+    excluded(timestamp),
+    unbounded
+  }
+  variant tstzbound {
+    included(timestamptz),
+    excluded(timestamptz),
+    unbounded
+  }
+
+  variant datebound {
+    included(date),
+    excluded(date),
+    unbounded
+  }
+
+  record int4range {
+   start: int4bound,
+   end: int4bound
+  }
+
+  record int8range {
+   start: int8bound,
+   end: int8bound
+  }
+
+  record numrange {
+   start: numbound,
+   end: numbound
+  }
+
+  record tsrange {
+   start: tsbound,
+   end: tsbound
+  }
+
+  record tstzrange {
+   start: tstzbound,
+   end: tstzbound
+  }
+
+  record daterange {
+   start: datebound,
+   end: datebound
+  }
+
+  record enumeration-type {
+    name: string
+  }
+
+  record enumeration {
+    name: string,
+    value: string
+  }
+
+  record composite-type {
+    name: string,
+    attributes: list<tuple<string, lazy-db-column-type>>
+  }
+
+  record composite {
+    name: string,
+    values: list<lazy-db-value>
+  }
+
+  record domain-type {
+    name: string,
+    base-type: lazy-db-column-type
+  }
+
+  record domain {
+    name: string,
+    value: lazy-db-value
+  }
+
+  variant db-column-type {
+    character,
+    int2,
+    int4,
+    int8,
+    float4,
+    float8,
+    numeric,
+    boolean,
+    text,
+    varchar,
+    bpchar,
+    timestamp,
+    timestamptz,
+    date,
+    time,
+    timetz,
+    interval,
+    bytea,
+    uuid,
+    xml,
+    json,
+    jsonb,
+    jsonpath,
+    inet,
+    cidr,
+    macaddr,
+    bit,
+    varbit,
+    int4range,
+    int8range,
+    numrange,
+    tsrange,
+    tstzrange,
+    daterange,
+    money,
+    oid,
+    enumeration(enumeration-type),
+    composite(composite-type),
+    domain(domain-type),
+    array(lazy-db-column-type)
+  }
+
+  variant db-value {
+    character(s8),
+    int2(s16),
+    int4(s32),
+    int8(s64),
+    float4(f32),
+    float8(f64),
+    numeric(string),
+    boolean(bool),
+    text(string),
+    varchar(string),
+    bpchar(string),
+    timestamp(timestamp),
+    timestamptz(timestamptz),
+    date(date),
+    time(time),
+    timetz(timetz),
+    interval(interval),
+    bytea(list<u8>),
+    json(string),
+    jsonb(string),
+    jsonpath(string),
+    xml(string),
+    uuid(uuid),
+    inet(ip-address),
+    cidr(ip-address),
+    macaddr(mac-address),
+    bit(list<bool>),
+    varbit(list<bool>),
+    int4range(int4range),
+    int8range(int8range),
+    numrange(numrange),
+    tsrange(tsrange),
+    tstzrange(tstzrange),
+    daterange(daterange),
+    money(s64),
+    oid(u32),
+    enumeration(enumeration),
+    composite(composite),
+    domain(domain),
+    array(list<lazy-db-value>),
+    null
+  }
+
+  resource lazy-db-value {
+    create: static func(value: db-value) -> lazy-db-value;
```

Comment:
You can use `constructor` here to make it the primary and anonymous constructor, rather than having a "named" constructor like you do. This could lead to more idiomatic usage depending on the language.